server/aws-lsp-codewhisperer/src/language-server/chat/agents/docGeneration/constants.ts

```diff
@@ -0,0 +1,162 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MynahIcons, Status } from '@aws/mynah-ui'
+import { FollowUpTypes, NewFileInfo } from '../shared/types'
+import { i18n } from '../shared/i18n'
+
+// For uniquely identifiying which chat messages should be routed to Doc
+export const docChat = 'docChat'
+
+export const docScheme = 'aws-doc'
+
+export const featureName = 'Amazon Q Doc Generation'
+
+export function getFileSummaryPercentage(input: string): number {
+    // Split the input string by newline characters
+    const lines = input.split('\n')
+
+    // Find the line containing "summarized:"
+    const summaryLine = lines.find(line => line.includes('summarized:'))
+
+    // If the line is not found, return null
+    if (!summaryLine) {
+        return -1
+    }
+
+    // Extract the numbers from the summary line
+    const [summarized, total] = summaryLine.split(':')[1].trim().split(' of ').map(Number)
+
+    // Calculate the percentage
+    const percentage = (summarized / total) * 100
+
+    return percentage
+}
+
+const checkIcons = {
+    wait: '&#9744;',
+    current: '&#9744;',
+    done: '&#9745;',
+}
+
+const getIconForStep = (targetStep: number, currentStep: number) => {
+    return currentStep === targetStep
+        ? checkIcons.current
+        : currentStep > targetStep
+          ? checkIcons.done
+          : checkIcons.wait
+}
+
+export enum DocGenerationStep {
+    UPLOAD_TO_S3,
+    SUMMARIZING_FILES,
+    GENERATING_ARTIFACTS,
+}
+
+export const docGenerationProgressMessage = (currentStep: DocGenerationStep, mode: Mode) => `
+${mode === Mode.CREATE ? i18n('AWS.amazonq.doc.answer.creating') : i18n('AWS.amazonq.doc.answer.updating')}
+
+${getIconForStep(DocGenerationStep.UPLOAD_TO_S3, currentStep)} ${i18n('AWS.amazonq.doc.answer.scanning')}
+
+${getIconForStep(DocGenerationStep.SUMMARIZING_FILES, currentStep)} ${i18n('AWS.amazonq.doc.answer.summarizing')}
+
+${getIconForStep(DocGenerationStep.GENERATING_ARTIFACTS, currentStep)} ${i18n('AWS.amazonq.doc.answer.generating')}
+
+
+`
+
+export const docGenerationSuccessMessage = (mode: Mode) =>
+    mode === Mode.CREATE ? i18n('AWS.amazonq.doc.answer.readmeCreated') : i18n('AWS.amazonq.doc.answer.readmeUpdated')
+
+export const docRejectConfirmation = 'Your changes have been discarded.'
+
+export const FolderSelectorFollowUps = [
+    {
+        icon: 'ok' as MynahIcons,
+        pillText: 'Yes',
+        prompt: 'Yes',
+        status: 'success' as Status,
+        type: FollowUpTypes.ProceedFolderSelection,
```

Comment:
Presumably these are going to change, but for now we copied in their dependencies so this can build